src/classic/element/ReactElementValidator.js

```diff
@@ -361,26 +362,33 @@ var ReactElementValidator = {
     }
 
     if (type) {
-      var name = type.displayName || type.name;
-      if (type.propTypes) {
+      // Extract the component class from the element. Converts string types
+      // to a composite class which may have propTypes.
+      // TODO: Validating a string's propTypes is not decoupled from the
+      // rendering target which is problematic.
```

Comment:
Yes, but it's problematic if there are multiple output environments. E.g. if an ART subtree was defined in terms of string based target nodes. Then we don't know which of the multiple environments this particular string will end up at.

Another case is HTML/SVG where the `<title />` element means different things depending on which parent it ends up in. So they should have different propTypes depending on the parent. We don't really have a good way to solve this issue.